tests/shared/utils.spec.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const m = require('mochainon');
+const utils = require('../../lib/shared/utils');
+
+describe('Shared: Utils', function() {
+
+  describe('.makeFlatStartCaseObject(object)', function() {
```

Comment:
There are many things to test here. Some from the top of my head:

- Environment variable keys

For example: `ETCHER_DISABLE_UPDATES` should remain the same.

- Environment variables inside objects

For example:

```json
{
   "foo": {
      "FOO_BAR_BAZ": 3
   }
}
```

Should be transformed to `"Foo FOO_BAR_BAZ": 3`.

- Top level keys
- Keys that start with a number
- Keys that are already in start case (e.g: should not be modified)
- Keys containing spaces
- Keys whose values are arrays
- Arrays containing objects

There are also a set of classic values you can test for:

- An empty object
- Something that is not an object, like a string, or an array
- `null`